### PR TITLE
fix: support Check on userset types of users

### PR DESCRIPTION
**Pull request by jon-whit**
## Description
This fixes Checks so that they support evaluation of relationships on usersets. For example, `Check(folder:folder1, viewer, group:engineering#member)`. Prior to this PR checks of this form would always return false, because we did not support checks on userset types.

## Review Checklist
- [X] I have added documentation for new/changed functionality in this PR or in [openfga.dev](https://github.com/openfga/openfga.dev)
- [X] The correct base branch is being used, if not `main`

### CODE REVIEW

**Review by craigpastro — COMMENTED** [review]
I think contextual tuples may need some changes. And can we add tests to check as well please?

**Comment by jon-whit** [comment]
> I think contextual tuples may need some changes. And can we add tests to check as well please?

@craigpastro it seems to work fine with ContextualTuples. I added two tests in `server/test/check` for these cases. Let me know what you think!

File: storage/postgres/postgres.go [hunk 1]
```diff
@@ -236,8 +236,9 @@ func (p *Postgres) ReadUserTuple(ctx context.Context, store string, tupleKey *op
 	defer span.End()
 
 	objectType, objectID := tupleUtils.SplitObject(tupleKey.GetObject())
-	row := p.pool.QueryRow(ctx, `SELECT object_type, object_id, relation, _user FROM tuple WHERE store = $1 AND object_type = $2 AND object_id = $3 AND relation = $4 AND _user = $5 AND user_type = $6`,
-		store, objectType, objectID, tupleKey.GetRelation(), tupleKey.GetUser(), tupleUtils.User)
+	row := p.pool.QueryRow(ctx, `SELECT object_type, object_id, relation, _user FROM tuple WHERE store = $1 AND object_type = $2 AND object_id = $3 AND relation = $4 AND _user = $5`,
```
Comment (elbuo8) [inline storage/postgres/postgres.go:239]: Usertype should still be provided but extracted from the tupleKey. This way we guarantee we land in the correct Index https://github.com/openfga/openfga/blob/b7d987f12a5e8f093a7eefa52bfddea349c9f966/assets/migrations/postgres/001_initialize_schema.sql#L14-L16

Comment (jon-whit) [reply storage/postgres/postgres.go]: Good call. I've updated it to use the `user_type` column as a filter. Let me know how that looks.

**Review by elbuo8 — APPROVED** [review]

**Pull request merged**